hub/models.py

```diff
@@ -1382,16 +1367,21 @@ async def deferred_refresh_many(
         external_data_source: ExternalDataSource = await cls.objects.aget(
             id=external_data_source_id
         )
-        if external_data_source.auto_update_enabled:
-            await external_data_source.refresh_many(member_ids=member_ids)
+        await external_data_source.refresh_many(member_ids=member_ids)
 
     @classmethod
     async def deferred_refresh_all(cls, external_data_source_id: str):
         external_data_source: ExternalDataSource = await cls.objects.aget(
             id=external_data_source_id
         )
-        if external_data_source.auto_update_enabled:
-            await external_data_source.refresh_all()
+
+        members = await external_data_source.fetch_all()
+        batches = batched(members, settings.IMPORT_UPDATE_ALL_BATCH_SIZE)
+        for batch in batches:
+            member_ids = [
+                external_data_source.get_record_id(member) for member in batch
+            ]
+            await external_data_source.schedule_refresh_many(member_ids)
```

Comment:
So this doesn't actually run the fetch, it only creates an iterator? I'm trying to figure out how the IDs are accessible in the loop without them all having been fetched up front.

Related to this, perhaps it makes sense to save these fetched values to the database whilst we have them in memory, so that `refresh_many` can reuse then for mapping... But that's a sidequest that could be backlogged.

If nothing else, we could perhaps demote this method to the `AirtableSource` class, as it doesn't sound like normal behaviour for other CRMs. But happy to do that when we actually start implementing a second one.

Reply:
it creates an iterator which gets the IDs when it is iterated (`for batch in batches` and `for member in batch`)